app/src/interfaces/presentation-links/presentation-links.vue

```diff
@@ -15,65 +14,97 @@ type Link = {
 	url?: string;
 };
 
+type ParsedLink = Omit<Link, 'url'> & {
+	to?: string;
+	href?: string;
+};
+
 type Props = {
 	links: Link[];
 	collection: string;
-	primaryKey: string;
+	primaryKey?: PrimaryKey;
 };
 
 const props = withDefaults(defineProps<Props>(), {
 	links: () => [],
 });
 
+const api = useApi();
 const values = inject('values', ref<Record<string, any>>({}));
+const resolvedRelationalValues = ref<Record<string, any>>({});
+const { primaryKey } = toRefs(props);
+
+/**
+ * Get all deduplicated relational fields from the link-templates.
+ * For example:
+ * [ "related.field", "languages.code" ]
+ */
+function getRelatedFieldsFromTemplates() {
+	const allFields = props.links?.flatMap((link) => (!link.url ? [] : getFieldsFromTemplate(link.url)));
+
+	return (
+		[...new Set(allFields)]
+			// filter out non-relations, since they should be included in the values already
+			.filter((value) => value.includes('.'))
+	);
+}
 
-const { collection, primaryKey } = toRefs(props);
-
-const query = computed(() => {
-	const fields = new Set();
-
-	props.links.forEach((link) => {
-		getFieldsFromTemplate(link.url ?? '').forEach((field) => fields.add(field));
-	});
-
-	return {
-		fields: Array.from(fields),
-	} as Query;
-});
-
-const { item } = useItem(collection, primaryKey, query);
-const { fields } = useCollection(collection);
-
-const fullItem = computed(() => {
-	const itemValue = item.value ?? {};
-
-	for (const field of fields.value) {
-		if (
-			field.meta?.special?.some((special) => RELATIONAL_TYPES.includes(special as (typeof RELATIONAL_TYPES)[number]))
-		) {
-			continue;
-		}
-
-		itemValue[field.field] = values.value[field.field];
-	}
-
-	return itemValue;
-});
-
-const linksParsed = computed(() => {
-	return props.links.map((link) => {
-		const parsedLink = omit<Record<string, any>>(link, ['url']);
-		const linkValue = render(link.url ?? '', fullItem.value ?? {});
-
-		if (linkValue.startsWith('/')) {
-			parsedLink.to = linkValue;
-		} else {
-			parsedLink.href = linkValue;
+watch(
+	primaryKey,
+	async (value) => {
+		if (!value || value === '+') return;
+
+		const relatedFieldsFromTemplates = getRelatedFieldsFromTemplates();
+		if (relatedFieldsFromTemplates.length === 0) return;
+
+		try {
+			const response = await api.get(`${getEndpoint(props.collection)}/${value}`, {
+				params: {
+					fields: relatedFieldsFromTemplates,
+				},
+			});
+
+			// Pick only non-arrays because we cant render those types of relations
+			// For example a M2M relation will return an array
+			resolvedRelationalValues.value = pickBy(response.data.data, (value) => !Array.isArray(value));
+		} catch (err) {
+			unexpectedError(err);
 		}
-
-		return parsedLink;
-	});
-});
+	},
+	// Immediate for fetching when opening a new tab directly
+	{ immediate: true },
+);
+
+const linksParsed = computed(
+	() =>
+		props.links?.map((link) => {
```

Comment:
```suggestion
		props.links.map((link) => {
```
according to its types `props.link` cannot be undefined/null